javaparser-core/src/main/java/com/github/javaparser/ast/stmt/TryStmt.java

```diff
@@ -57,7 +58,7 @@
  */
 public final class TryStmt extends Statement {
 
-    private NodeList<VariableDeclarationExpr> resources;
+    private NodeList<Expression> resources;
```

Comment:
Is this change specific for Java 9?
Looking here: https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
It says: "The try-with-resources statement is a try statement that declares one or more resources".
Should we need to guide users to use only VariableDeclarationExpr unless they are using Java 9?

Reply:
Something like adding a validation that adds a Problem when parsing with Java 8 and before and we find something that is not a variable decl? Sure.

Reply:
Added.